board/views.py

```diff
@@ -111,3 +123,39 @@ def get(self, req, image):
         """Gets the requested image from the specified image URI."""
         #TODO: return image blob and fix docstring once finished
         return HttpResponse(f'<h1>wow u got image {image}</h1>')
+
+
+class GetBoardDetails(View):
+    """
+    The API endpoint to retrieve details regarding the board page.
+
+    The details DO NOT include the posts, as they are done through a separate API.
+    This returns a JSON object in the form of:
+
+    JSON fields:
+        title -> `string`: the title of the board. 
+        description -> `string`: the description for the board.
+        bg -> `string`: the uuid of the background image of the board. 
+
+    """
+
+    def get(self, req):
+        """Get the board details as a JSON response."""
+        
+        try:
+            # Ensure the query string exists.
+            board_uuid = req.GET.get('board')
+            # Ensure it is a valid uuid.
+            UUID(board_uuid, version=4)
+            # Ensure the board exists.
+            board = Board.objects.get(uuid=board_uuid)
+        except:
+            raise Http404('Board not found')
```

Comment:
In most apps, are 404s generally raised if the UUID is malformed?

Reply:
Just off the top of my head I would assume that something like bad request 400 would be raised but I suppose to the user it's all the same either way

I would assume 404 is if UUID is present and not malformed but a board cannot be found.

Reply:
`In an API, this can also mean that the endpoint is valid but the resource itself does not exist.`

Malformed is still `resource doesn't exist`.